Parse/ParseClientConfiguration.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Parse/PFConstants.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol ParseMutableClientConfiguration<NSObject>
+
+@property (nonatomic, nullable, copy) NSString *applicationId;
+@property (nonatomic, nullable, copy) NSString *clientKey;
+
+@property (nonatomic, assign, getter=isLocalDatastoreEnabled) BOOL localDatastoreEnabled;
+
+@property (nonatomic, nullable, copy) NSString *applicationGroupIdentifier;
+@property (nonatomic, nullable, copy) NSString *containingApplicationBundleIdentifier;
+
+@end
+
+/*!
+ This represents the entirety of a parse configuration
+ */
+@interface ParseClientConfiguration : NSObject<NSCopying>
+
++ (instancetype)new NS_UNAVAILABLE;
++ (instancetype)alloc NS_UNAVAILABLE;
++ (instancetype)allocWithZone:(NSZone *)zone NS_UNAVAILABLE;
+- (instancetype)init NS_UNAVAILABLE;
+
+@property (nonatomic, readonly, copy) NSString *applicationId;
+@property (nonatomic, readonly, copy) NSString *clientKey;
+
+@property (nonatomic, readonly, assign, getter=isLocalDatastoreEnabled) BOOL localDatastoreEnabled;
```

Comment:
Order is as follows (for consistency):
`(nullable/null_unspecified/null_resettable), nonatomic/atomic, assign/strong/copy/retain, (readonly/readwrite), (getter=...), (setter=...)`